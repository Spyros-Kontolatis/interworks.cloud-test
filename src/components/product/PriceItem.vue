<template>
  <div>{{ formattedPrice }}</div>
</template>

<script setup lang="ts">
import { defineProps, computed } from "vue";
import { useCurrency } from "@/composables/useCurrency";

const props = defineProps({
  price: {
    type: Number,
    required: true,
  },
});

const formattedPrice = computed(() => useCurrency(props.price));
</script>
