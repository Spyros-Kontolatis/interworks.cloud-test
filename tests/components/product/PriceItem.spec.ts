import { shallowMount } from "@vue/test-utils";
import PriceItem from "../../../src/components/product/PriceItem.vue";

describe("PriceItem.vue", () => {
  it("renders the component", () => {
    const wrapper = shallowMount(PriceItem, {
      props: {
        price: 2,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
