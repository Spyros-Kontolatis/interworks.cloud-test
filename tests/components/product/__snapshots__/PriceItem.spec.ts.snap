// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceItem.vue renders the component 1`] = `
<div>
  2.00â‚¬
</div>
`;
